### PR TITLE
SASS-1669 - Save and continue for expenses

### PR DESCRIPTION
**Pull request by HarryGull**
Add 'save and continue' functionality to button on 'check your employment expenses' page

https://jira.tools.tax.service.gov.uk/browse/SASS-1669

##### Before Reviewing
- [x]  Have you pulled the branch down?
- [x]  Have you assigned yourself to the PR?
- [x]  Have you moved the task to “in review” on JIRA?
- [x]  Have you checked to ensure all dependencies are up to date?
- [x]  Have you checked to ensure its been rebased against the current version of main?

##### Whilst Reviewing
- [x]  Have you run the tests?
- [x]  Have you run the journey tests?
- [x]  Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?

##### After Reviewing
- [x]  Have you checked for merge conflicts or any changes in the current main that may affect the current pull request? i.e. does it need another rebase?
- [x]  Have you checked to make sure there are no builds in the pipeline before you merge?
- [x]  Have you moved the task to “in pipeline” on Jira?

##### Before creating PR
- [x]  Have you run the tests?
- [x]  Have you run the journey tests? (where applicable)
- [x]  Have you addressed warnings where appropriate?
- [x]  Have you rebased against the current version of main?
- [x]  Have you checked code coverage isn’t lower than previously?

##### After PRs been raised
- [ ]  Have you checked the PR Builder passes?

### CODE REVIEW

File: app/connectors/httpParsers/NrsSubmissionHttpParser.scala [hunk 1]
```diff
@@ -43,4 +43,4 @@ object NrsSubmissionHttpParser extends APIParser {
       }
     }
   }
-}
\ No newline at end of file
+}
```
Comment (abiodxn) [inline app/connectors/httpParsers/NrsSubmissionHttpParser.scala:46]: added empty line?

File: app/controllers/employment/CheckEmploymentDetailsController.scala [hunk 2]
```diff
@@ -24,7 +24,8 @@ import controllers.employment.routes.CheckEmploymentDetailsController
 import controllers.predicates.{AuthorisedAction, InYearAction}
 import models.User
 import models.employment.createUpdate.CreateUpdateEmploymentRequest
-import models.employment.{AllEmploymentData, DecodedAmendEmploymentDetailsPayload, DecodedCreateNewEmploymentDetailsPayload, DecodedPriorEmploymentInfo, EmploymentDetailsViewModel}
+import models.employment.{AllEmploymentData, DecodedAmendEmploymentDetailsPayload, DecodedCreateNewEmploymentDetailsPayload,
+  DecodedPriorEmploymentInfo, EmploymentDetailsViewModel}
```
Comment (abiodxn) [inline app/controllers/employment/CheckEmploymentDetailsController.scala:28]: too many imports just import all using _

File: app/models/employment/createUpdate/CreateUpdateEmploymentRequest.scala [hunk 3]
```diff
@@ -19,7 +19,8 @@ package models.employment.createUpdate
 import audit.{AmendEmploymentDetailsUpdateAudit, AuditEmploymentData, AuditNewEmploymentData, CreateNewEmploymentDetailsAudit, PriorEmploymentAuditInfo}
 import models.User
 import models.benefits.Benefits
-import models.employment.{DecodedAmendEmploymentDetailsPayload, DecodedCreateNewEmploymentDetailsPayload, DecodedEmploymentData, DecodedNewEmploymentData, DecodedPriorEmploymentInfo, Deductions, EmploymentSource}
+import models.employment.{DecodedAmendEmploymentDetailsPayload, DecodedCreateNewEmploymentDetailsPayload, DecodedEmploymentData,
+  DecodedNewEmploymentData, DecodedPriorEmploymentInfo, Deductions, EmploymentSource}
```
Comment (abiodxn) [inline app/models/employment/createUpdate/CreateUpdateEmploymentRequest.scala:23]: import all

File: app/models/employment/DecodedCreateNewEmploymentDetailsPayload.scala [hunk 4]
```diff
@@ -31,4 +31,4 @@ case class DecodedCreateNewEmploymentDetailsPayload(employmentData: DecodedNewEm
 
 object DecodedCreateNewEmploymentDetailsPayload {
   implicit def writes: OWrites[DecodedCreateNewEmploymentDetailsPayload] = Json.writes[DecodedCreateNewEmploymentDetailsPayload]
-}
\ No newline at end of file
+}
```
Comment (abiodxn) [inline app/models/employment/DecodedCreateNewEmploymentDetailsPayload.scala:34]: empty line

File: it/connectors/CreateOrAmendExpensesConnectorSpec.scala [hunk 5]
```diff
@@ -18,7 +18,8 @@ package connectors
 
 import com.github.tomakehurst.wiremock.http.HttpHeader
 import connectors.httpParsers.CreateOrAmendExpensesHttpParser.CreateOrAmendExpensesResponse
-import models.expenses.CreateExpensesRequestModel
+import models.expenses.createUpdate
+import models.expenses.createUpdate.CreateUpdateExpensesRequest
 import models.{APIErrorBodyModel, APIErrorModel}
```
Comment (abiodxn) [inline it/connectors/CreateOrAmendExpensesConnectorSpec.scala:23]: do we need to have both imports ?

File: it/connectors/DeleteOrIgnoreExpensesConnectorSpec.scala [hunk 6]
```diff
@@ -18,10 +18,10 @@ package connectors
 
 import com.github.tomakehurst.wiremock.http.HttpHeader
 import connectors.httpParsers.CreateOrAmendExpensesHttpParser.CreateOrAmendExpensesResponse
-import models.expenses.CreateExpensesRequestModel
+import models.expenses.createUpdate
+import models.expenses.createUpdate.CreateUpdateExpensesRequest
```
Comment (abiodxn) [inline it/connectors/DeleteOrIgnoreExpensesConnectorSpec.scala:22]: again do we need both. I think we can just use one ?

**Review by abiodxn — CHANGES_REQUESTED** [review]

File: app/connectors/httpParsers/NrsSubmissionHttpParser.scala (hunk 1 above)
Comment (HarryGull) [reply app/connectors/httpParsers/NrsSubmissionHttpParser.scala:46]: Yes that was deliberate. I added a few like that to remove the Scala style warnings that specify the rule the end of file should have an empty line.

File: app/controllers/employment/CheckEmploymentDetailsController.scala (hunk 2 above)
Comment (HarryGull) [reply app/controllers/employment/CheckEmploymentDetailsController.scala]: OK

File: app/models/employment/createUpdate/CreateUpdateEmploymentRequest.scala (hunk 3 above)
Comment (HarryGull) [reply app/models/employment/createUpdate/CreateUpdateEmploymentRequest.scala]: ok

File: it/connectors/DeleteOrIgnoreExpensesConnectorSpec.scala (hunk 6 above)
Comment (HarryGull) [reply it/connectors/DeleteOrIgnoreExpensesConnectorSpec.scala]: I can’t see this. Maybe it was already removed by my rebase as there is only one?

File: it/connectors/CreateOrAmendExpensesConnectorSpec.scala (hunk 5 above)
Comment (HarryGull) [reply it/connectors/CreateOrAmendExpensesConnectorSpec.scala]: I can’t see this. Maybe it was already removed by my rebase as there is only one?

File: app/models/employment/DecodedCreateNewEmploymentDetailsPayload.scala (hunk 4 above)
Comment (HarryGull) [reply app/models/employment/DecodedCreateNewEmploymentDetailsPayload.scala:34]: Yes that was deliberate. I added a few like that to remove the Scala style warnings that specify the rule the end of file should have an empty line.

I can’t see this. Maybe it was already removed by my rebase as there is only one?

**Review by abiodxn — APPROVED** [review]

**Pull request merged**